spec/lib/castle/extractors/ip_spec.rb

```diff
@@ -1,21 +1,84 @@
 # frozen_string_literal: true
 
 describe Castle::Extractors::IP do
-  subject(:extractor) { described_class.new(request) }
+  subject(:extractor) { described_class.new(formatted_headers) }
 
+  let(:formatted_headers) { Castle::HeaderFilter.new(request).call }
   let(:request) { Rack::Request.new(env) }
 
   describe 'ip' do
     context 'when regular ip' do
+      let(:env) { Rack::MockRequest.env_for('/', 'HTTP_X_FORWARDED_FOR' => '1.2.3.5') }
```

Comment:
I'd prefer to remove env and request from this spec and just pass properly formatted headers instead